In [1]:
import pandas as pd

In [2]:
evals = pd.read_excel("data/BDI3 All Evals for NSS 11-2023.xlsx")

In [3]:
evals

Child ID Gender Date of Birth Location - Sub Level 1  \
0        44879      F    2022-09-27              Southwest   
1        47795      F    2020-12-23      Greater Nashville   
2        54340      M    2021-06-22        First Tennessee   
3        54344      F    2022-03-07    Southeast Tennessee   
4        54515      M    2021-11-04          Memphis Delta   
...        ...    ...           ...                    ...   
23942  SP03287      F    2023-09-25          Memphis Delta   
23943  SP03288      F    2021-05-18          South Central   
23944  SP03289      F    2021-10-31          South Central   
23945  SP03290      M    2020-03-25          Memphis Delta   
23946  SP03291      M    2023-09-18      Greater Nashville   

                      Program Label  Adaptive Sum of Scaled Scores  \
0      BDI-3 Eligibility Evaluation                            7.0   
1      BDI-3 Eligibility Evaluation                           16.0   
2      BDI-3 Eligibility Evaluation                            8.0   
3      BDI-3 Eligibility Evaluation                           13.0   
4      BDI-3 Eligibility Evaluation                            5.0   
...                             ...                            ...   
23942  BDI-3 Eligibility Evaluation                           13.0   
23943  BDI-3 Eligibility Evaluation                           25.0   
23944  BDI-3 Eligibility Evaluation                           21.0   
23945       BDI-3 Annual Evaluation                           16.0   
23946  BDI-3 Eligibility Evaluation                           11.0   

       Adaptive Developmental Quotient Adaptive Percentile Rank  \
0                                 85.0                       16   
1                                 88.0                       21   
2                                 90.0                       25   
3                                115.0                       84   
4                                 75.0                        5   
...                                ...                      ...   
23942                            115.0                       84   
23943                            115.0                       84   
23944                            103.0                       58   
23945                             88.0                       21   
23946                            105.0                       63   

      Adaptive 95% Confidence Interval         Adaptive RDI  ... Code 1  \
0                                77-97                39/90  ...    NaN   
1                                82-96                79/90  ...    NaN   
2                               81-101                77/90  ...    NaN   
3                              103-123                99/90  ...    NaN   
4                                68-88  1990-09-01 00:00:00  ...    NaN   
...                                ...                  ...  ...    ...   
23942                          103-123                99/90  ...    NaN   
23943                          107-121                96/90  ...    NaN   
23944                           96-110                92/90  ...    NaN   
23945                            82-96                66/90  ...    NaN   
23946                           94-114                94/90  ...    NaN   

       Code 2 Code 3  Code 4  Code 5 Code 6 Code 7 Code 8 Code 9  Code 10  
0         NaN    NaN     NaN     NaN    NaN    NaN    NaN    NaN      NaN  
1         NaN    NaN     NaN     NaN    NaN    NaN    NaN    NaN      NaN  
2         NaN    NaN     NaN     NaN    NaN    NaN    NaN    NaN      NaN  
3         NaN    NaN     NaN     NaN    NaN    NaN    NaN    NaN      NaN  
4         NaN    NaN     NaN     NaN    NaN    NaN    NaN    NaN      NaN  
...       ...    ...     ...     ...    ...    ...    ...    ...      ...  
23942     NaN    NaN     NaN     NaN    NaN    NaN    NaN    NaN      NaN  
23943     NaN    NaN     NaN     NaN    NaN    NaN    NaN    NaN      NaN  
23944     NaN    NaN     NaN  

In [4]:
evals = evals.drop(columns=['Code 1', 'Code 2', 'Code 3', 'Code 4', 'Code 5', 'Code 6', 'Code 7', 'Code 8', 'Code 9', 'Code 10'])

In [5]:
evals.head(2)

Child ID Gender Date of Birth Location - Sub Level 1  \
0    44879      F    2022-09-27              Southwest   
1    47795      F    2020-12-23      Greater Nashville   

                  Program Label  Adaptive Sum of Scaled Scores  \
0  BDI-3 Eligibility Evaluation                            7.0   
1  BDI-3 Eligibility Evaluation                           16.0   

   Adaptive Developmental Quotient Adaptive Percentile Rank  \
0                             85.0                       16   
1                             88.0                       21   

  Adaptive 95% Confidence Interval Adaptive RDI  ...  \
0                            77-97        39/90  ...   
1                            82-96        79/90  ...   

  Cognitive-Perception and Concepts RS  Cognitive-Perception and Concepts SS  \
0                                  9.0                                  13.0   
1                                 11.0                                   7.0   

  Cognitive-Perception and Concepts PR  Cognitive-Perception and Concepts AE  \
0                                   84                                    12   
1                                   16                                    18   

   Cognitive-Perception and Concepts RDI  \
0                                  98/90   
1                                  29/90   

  Cognitive-Perception and Concepts CSS  \
0                                 411.0   
1                                 437.0   

  Cognitive-Perception and Concepts CSS 90%  \
0                                 392 - 431   
1                                 421 - 453   

  Cognitive-Perception and Concepts Z-Score  \
0                                         1   
1                                    – 1.00   

  Cognitive-Perception and Concepts T-Score  \
0                                      60.0   
1                                      40.0   

   Cognitive-Perception and Concepts NCE  
0                                     71  
1                                     29  

[2 rows x 209 columns]

In [6]:
len(evals.loc[evals['Gender'] == 'F'])

8801

In [7]:
len(evals.loc[evals['Gender'] == 'M'])

14732

In [8]:
len(evals)

23947

there are 414 missing...whats the other gender

In [9]:
evals.Gender.unique()

array(['F', 'M', 'U'], dtype=object)

In [10]:
len(evals.loc[evals['Gender'] == 'U'])

414

In [11]:
from datetime import datetime

In [12]:
import numpy as np

make an age column!

In [13]:
evals['Age'] = np.floor((datetime.now() - evals['Date of Birth']).dt.days / 365.25)

In [14]:
evals.head(2)

Child ID Gender Date of Birth Location - Sub Level 1  \
0    44879      F    2022-09-27              Southwest   
1    47795      F    2020-12-23      Greater Nashville   

                  Program Label  Adaptive Sum of Scaled Scores  \
0  BDI-3 Eligibility Evaluation                            7.0   
1  BDI-3 Eligibility Evaluation                           16.0   

   Adaptive Developmental Quotient Adaptive Percentile Rank  \
0                             85.0                       16   
1                             88.0                       21   

  Adaptive 95% Confidence Interval Adaptive RDI  ...  \
0                            77-97        39/90  ...   
1                            82-96        79/90  ...   

  Cognitive-Perception and Concepts SS  Cognitive-Perception and Concepts PR  \
0                                 13.0                                    84   
1                                  7.0                                    16   

  Cognitive-Perception and Concepts AE  Cognitive-Perception and Concepts RDI  \
0                                   12                                  98/90   
1                                   18                                  29/90   

   Cognitive-Perception and Concepts CSS  \
0                                  411.0   
1                                  437.0   

  Cognitive-Perception and Concepts CSS 90%  \
0                                 392 - 431   
1                                 421 - 453   

  Cognitive-Perception and Concepts Z-Score  \
0                                         1   
1                                    – 1.00   

  Cognitive-Perception and Concepts T-Score  \
0                                      60.0   
1                                      40.0   

  Cognitive-Perception and Concepts NCE  Age  
0                                    71  1.0  
1                                    29  2.0  

[2 rows x 210 columns]

In [15]:
evals.Age.describe()

count    23947.000000
mean         1.873304
std          0.953605
min          0.000000
25%          1.000000
50%          2.000000
75%          3.000000
max          4.000000
Name: Age, dtype: float64

In [16]:
evals.columns = [c.replace(' ', '_') for c in evals.columns]

In [17]:
evals.Adaptive_Sum_of_Scaled_Scores.describe()

count    23939.000000
mean        11.728268
std          6.001474
min          1.000000
25%          7.000000
50%         11.000000
75%         16.000000
max         36.000000
Name: Adaptive_Sum_of_Scaled_Scores, dtype: float64

In [18]:
evals.Adaptive_Percentile_Rank.describe()

count     23939
unique       37
top          16
freq       2038
Name: Adaptive_Percentile_Rank, dtype: int64

In [19]:
columns = list(evals.columns.values)

In [20]:
columns

['Child_ID',
 'Gender',
 'Date_of_Birth',
 'Location_-_Sub_Level_1',
 'Program_Label',
 'Adaptive_Sum_of_Scaled_Scores',
 'Adaptive_Developmental_Quotient',
 'Adaptive_Percentile_Rank',
 'Adaptive_95%_Confidence_Interval',
 'Adaptive_RDI',
 'Adaptive_Z-Score',
 'Adaptive_T-Score',
 'Adaptive_NCE',
 'Social-Emotional_Sum_of_Scaled_Scores',
 'Social-Emotional_Developmental_Quotient',
 'Social-Emotional_Percentile_Rank',
 'Social-Emotional_95%_Confidence_Interval',
 'Social-Emotional_RDI',
 'Social-Emotional_Z-Score',
 'Social-Emotional_T-Score',
 'Social-Emotional_NCE',
 'Communication_Sum_of_Scaled_Scores',
 'Communication_Developmental_Quotient',
 'Communication_Percentile_Rank',
 'Communication_95%_Confidence_Interval',
 'Communication_RDI',
 'Communication_Z-Score',
 'Communication_T-Score',
 'Communication_NCE',
 'Motor_Sum_of_Scaled_Scores',
 'Motor_Developmental_Quotient',
 'Motor_Percentile_Rank',
 'Motor_95%_Confidence_Interval',
 'Motor_RDI',
 'Motor_Z-Score',
 'Motor_T-Score',

In [21]:
len(evals.Child_ID.unique())

19003

23947 is the number of rows...vs 19003 unique ids....(4944)

break the main df into smaller subsets based on each domain for better analysis

In [28]:
adaptive_domain = evals[['Child_ID',
 'Gender',
 'Date_of_Birth','Age',
 'Location_-_Sub_Level_1',
 'Program_Label',
 'Adaptive_Sum_of_Scaled_Scores',
 'Adaptive_Developmental_Quotient',
 'Adaptive_Percentile_Rank',
 'Adaptive_95%_Confidence_Interval',
 'Adaptive_RDI',
 'Adaptive_Z-Score',
 'Adaptive_T-Score',
 'Adaptive_NCE','Adaptive-Self_Care_Examiner',
 'Adaptive-Self_Care_Date_of_Testing',
 'Adaptive-Self_Care_RS',
 'Adaptive-Self_Care_SS',
 'Adaptive-Self_Care_PR',
 'Adaptive-Self_Care_AE',
 'Adaptive-Self_Care_RDI',
 'Adaptive-Self_Care_CSS',
 'Adaptive-Self_Care_CSS_90%',
 'Adaptive-Self_Care_Z-Score',
 'Adaptive-Self_Care_T-Score',
 'Adaptive-Self_Care_NCE',
 'Adaptive-Personal_Responsibility_Examiner',
 'Adaptive-Personal_Responsibility_Date_of_Testing',
 'Adaptive-Personal_Responsibility_RS',
 'Adaptive-Personal_Responsibility_SS',
 'Adaptive-Personal_Responsibility_PR',
 'Adaptive-Personal_Responsibility_AE',
 'Adaptive-Personal_Responsibility_RDI',
 'Adaptive-Personal_Responsibility_CSS',
 'Adaptive-Personal_Responsibility_CSS_90%',
 'Adaptive-Personal_Responsibility_Z-Score',
 'Adaptive-Personal_Responsibility_T-Score',
 'Adaptive-Personal_Responsibility_NCE']]

In [29]:
adaptive_domain.head(2)

Child_ID Gender Date_of_Birth  Age Location_-_Sub_Level_1  \
0    44879      F    2022-09-27  1.0              Southwest   
1    47795      F    2020-12-23  2.0      Greater Nashville   

                  Program_Label  Adaptive_Sum_of_Scaled_Scores  \
0  BDI-3 Eligibility Evaluation                            7.0   
1  BDI-3 Eligibility Evaluation                           16.0   

   Adaptive_Developmental_Quotient Adaptive_Percentile_Rank  \
0                             85.0                       16   
1                             88.0                       21   

  Adaptive_95%_Confidence_Interval  ... Adaptive-Personal_Responsibility_RS  \
0                            77-97  ...                                 NaN   
1                            82-96  ...                                 7.0   

  Adaptive-Personal_Responsibility_SS  Adaptive-Personal_Responsibility_PR  \
0                                 NaN                                  NaN   
1                                 9.0                                   37   

  Adaptive-Personal_Responsibility_AE Adaptive-Personal_Responsibility_RDI  \
0                                 NaN                                  NaN   
1                                 <24                                87/90   

  Adaptive-Personal_Responsibility_CSS  \
0                                  NaN   
1                                484.0   

   Adaptive-Personal_Responsibility_CSS_90%  \
0                                       NaN   
1                                 476 - 493   

   Adaptive-Personal_Responsibility_Z-Score  \
0                                       NaN   
1                                    – 0.33   

  Adaptive-Personal_Responsibility_T-Score  \
0                                      NaN   
1                                     47.0   

  Adaptive-Personal_Responsibility_NCE  
0                                  NaN  
1                                   43  

[2 rows x 38 columns]

In [27]:
socialemotional_domain = evals[['Child_ID',
 'Gender',
 'Date_of_Birth', 'Age',
 'Location_-_Sub_Level_1',
 'Program_Label', 'Social-Emotional_Sum_of_Scaled_Scores',
 'Social-Emotional_Developmental_Quotient',
 'Social-Emotional_Percentile_Rank',
 'Social-Emotional_95%_Confidence_Interval',
 'Social-Emotional_RDI',
 'Social-Emotional_Z-Score',
 'Social-Emotional_T-Score',
 'Social-Emotional_NCE', 'Social_Emotional-Adult_Interaction_Examiner',
 'Social_Emotional-Adult_Interaction_Date_of_Testing',
 'Social_Emotional-Adult_Interaction_RS',
 'Social_Emotional-Adult_Interaction_SS',
 'Social_Emotional-Adult_Interaction_PR',
 'Social_Emotional-Adult_Interaction_AE',
 'Social_Emotional-Adult_Interaction_RDI',
 'Social_Emotional-Adult_Interaction_CSS',
 'Social_Emotional-Adult_Interaction_CSS_90%',
 'Social_Emotional-Adult_Interaction_Z-Score',
 'Social_Emotional-Adult_Interaction_T-Score',
 'Social_Emotional-Adult_Interaction_NCE',
 'Social_Emotional-Peer_Interaction_Examiner',
 'Social_Emotional-Peer_Interaction_Date_of_Testing',
 'Social_Emotional-Peer_Interaction_RS',
 'Social_Emotional-Peer_Interaction_SS',
 'Social_Emotional-Peer_Interaction_PR',
 'Social_Emotional-Peer_Interaction_AE',
 'Social_Emotional-Peer_Interaction_RDI',
 'Social_Emotional-Peer_Interaction_CSS',
 'Social_Emotional-Peer_Interaction_CSS_90%',
 'Social_Emotional-Peer_Interaction_Z-Score',
 'Social_Emotional-Peer_Interaction_T-Score',
 'Social_Emotional-Peer_Interaction_NCE',
 'Social_Emotional-Self_Concept_/_Social_Role_Examiner',
 'Social_Emotional-Self_Concept_/_Social_Role_Date_of_Testing',
 'Social_Emotional-Self_Concept_/_Social_Role_RS',
 'Social_Emotional-Self_Concept_/_Social_Role_SS',
 'Social_Emotional-Self_Concept_/_Social_Role_PR',
 'Social_Emotional-Self_Concept_/_Social_Role_AE',
 'Social_Emotional-Self_Concept_/_Social_Role_RDI',
 'Social_Emotional-Self_Concept_/_Social_Role_CSS',
 'Social_Emotional-Self_Concept_/_Social_Role_CSS_90%',
 'Social_Emotional-Self_Concept_/_Social_Role_Z-Score',
 'Social_Emotional-Self_Concept_/_Social_Role_T-Score',
 'Social_Emotional-Self_Concept_/_Social_Role_NCE']]
socialemotional_domain.head(2)

Child_ID Gender Date_of_Birth  Age Location_-_Sub_Level_1  \
0    44879      F    2022-09-27  1.0              Southwest   
1    47795      F    2020-12-23  2.0      Greater Nashville   

                  Program_Label  Social-Emotional_Sum_of_Scaled_Scores  \
0  BDI-3 Eligibility Evaluation                                   29.0   
1  BDI-3 Eligibility Evaluation                                   32.0   

   Social-Emotional_Developmental_Quotient Social-Emotional_Percentile_Rank  \
0                                    127.0                               96   
1                                    104.0                               61   

  Social-Emotional_95%_Confidence_Interval  ...  \
0                                  118-132  ...   
1                                  100-108  ...   

  Social_Emotional-Self_Concept_/_Social_Role_RS  \
0                                           17.0   
1                                           32.0   

  Social_Emotional-Self_Concept_/_Social_Role_SS  \
0                                           15.0   
1                                           11.0   

   Social_Emotional-Self_Concept_/_Social_Role_PR  \
0                                              95   
1                                              63   

  Social_Emotional-Self_Concept_/_Social_Role_AE  \
0                                             14   
1                                             30   

  Social_Emotional-Self_Concept_/_Social_Role_RDI  \
0                                           98/90   
1                                           95/90   

  Social_Emotional-Self_Concept_/_Social_Role_CSS  \
0                                           464.0   
1                                           496.0   

   Social_Emotional-Self_Concept_/_Social_Role_CSS_90%  \
0                                          456 - 473     
1                                          489 - 503     

   Social_Emotional-Self_Concept_/_Social_Role_Z-Score  \
0                                               1.67     
1                                               0.33     

  Social_Emotional-Self_Concept_/_Social_Role_T-Score  \
0                                               67.0    
1                                               53.0    

  Social_Emotional-Self_Concept_/_Social_Role_NCE  
0                                              85  
1                                              57  

[2 rows x 50 columns]

In [30]:
communication_domain = evals[['Child_ID',
 'Gender',
 'Date_of_Birth','Age',
 'Location_-_Sub_Level_1',
 'Program_Label','Communication_Sum_of_Scaled_Scores',
 'Communication_Developmental_Quotient',
 'Communication_Percentile_Rank',
 'Communication_95%_Confidence_Interval',
 'Communication_RDI',
 'Communication_Z-Score',
 'Communication_T-Score',
 'Communication_NCE','Communication-Receptive_Communication_Examiner',
 'Communication-Receptive_Communication_Date_of_Testing',
 'Communication-Receptive_Communication_RS',
 'Communication-Receptive_Communication_SS',
 'Communication-Receptive_Communication_PR',
 'Communication-Receptive_Communication_AE',
 'Communication-Receptive_Communication_RDI',
 'Communication-Receptive_Communication_CSS',
 'Communication-Receptive_Communication_CSS_90%_',
 'Communication-Receptive_Communication_Z-Score',
 'Communication-Receptive_Communication_T-Score',
 'Communication-Receptive_Communication_NCE',
 'Communication-Expressive_Communication_Examiner',
 'Communication-Expressive_Communication_Date_of_Testing',
 'Communication-Expressive_Communication_RS',
 'Communication-Expressive_Communication_SS',
 'Communication-Expressive_Communication_PR',
 'Communication-Expressive_Communication_AE',
 'Communication-Expressive_Communication_RDI',
 'Communication-Expressive_Communication_CSS',
 'Communication-Expressive_Communication_CSS_90%_',
 'Communication-Expressive_Communication_Z-Score',
 'Communication-Expressive_Communication_T-Score',
 'Communication-Expressive_Communication_NCE']]
communication_domain.head(2)

Child_ID Gender Date_of_Birth  Age Location_-_Sub_Level_1  \
0    44879      F    2022-09-27  1.0              Southwest   
1    47795      F    2020-12-23  2.0      Greater Nashville   

                  Program_Label  Communication_Sum_of_Scaled_Scores  \
0  BDI-3 Eligibility Evaluation                                17.0   
1  BDI-3 Eligibility Evaluation                                19.0   

   Communication_Developmental_Quotient Communication_Percentile_Rank  \
0                                  91.0                            27   
1                                  97.0                            42   

  Communication_95%_Confidence_Interval  ...  \
0                                84-100  ...   
1                                92-102  ...   

  Communication-Expressive_Communication_RS  \
0                                       8.0   
1                                      28.0   

  Communication-Expressive_Communication_SS  \
0                                       9.0   
1                                       9.0   

   Communication-Expressive_Communication_PR  \
0                                         37   
1                                         37   

  Communication-Expressive_Communication_AE  \
0                                         7   
1                                        22   

  Communication-Expressive_Communication_RDI  \
0                                      81/90   
1                                      77/90   

  Communication-Expressive_Communication_CSS  \
0                                      396.0   
1                                      464.0   

   Communication-Expressive_Communication_CSS_90%_  \
0                                        386 - 406   
1                                        455 - 472   

   Communication-Expressive_Communication_Z-Score  \
0                                          – 0.33   
1                                          – 0.33   

  Communication-Expressive_Communication_T-Score  \
0                                           47.0   
1                                           47.0   

  Communication-Expressive_Communication_NCE  
0                                         43  
1                                         43  

[2 rows x 38 columns]

In [31]:
motor_domain = evals[['Child_ID',
 'Gender',
 'Date_of_Birth','Age',
 'Location_-_Sub_Level_1',
 'Program_Label','Motor_Sum_of_Scaled_Scores',
 'Motor_Developmental_Quotient',
 'Motor_Percentile_Rank',
 'Motor_95%_Confidence_Interval',
 'Motor_RDI',
 'Motor_Z-Score',
 'Motor_T-Score',
 'Motor_NCE','Motor-Gross_Motor_Examiner',
 'Motor-Gross_Motor_Date_of_Testing',
 'Motor-Gross_Motor_RS',
 'Motor-Gross_Motor_SS',
 'Motor-Gross_Motor_PR',
 'Motor-Gross_Motor_AE',
 'Motor-Gross_Motor_RDI',
 'Motor-Gross_Motor_CSS',
 'Motor-Gross_Motor_CSS_90%_',
 'Motor-Gross_Motor_Z-Score',
 'Motor-Gross_Motor_T-Score',
 'Motor-Gross_Motor_NCE',
 'Motor-Fine_Motor_Examiner',
 'Motor-Fine_Motor_Date_of_Testing',
 'Motor-Fine_Motor_RS',
 'Motor-Fine_Motor_SS',
 'Motor-Fine_Motor_PR',
 'Motor-Fine_Motor_AE',
 'Motor-Fine_Motor_RDI',
 'Motor-Fine_Motor_CSS',
 'Motor-Fine_Motor_CSS_90%',
 'Motor-Fine_Motor_Z-Score',
 'Motor-Fine_Motor_T-Score',
 'Motor-Fine_Motor_NCE',
 'Motor-Perceptual_Motor_Examiner',
 'Motor-Perceptual_Motor_Date_of_Testing',
 'Motor-Perceptual_Motor_RS',
 'Motor-Perceptual_Motor_SS',
 'Motor-Perceptual_Motor_PR',
 'Motor-Perceptual_Motor_AE',
 'Motor-Perceptual_Motor_RDI',
 'Motor-Perceptual_Motor_CSS',
 'Motor-Perceptual_Motor_CSS_90%_',
 'Motor-Perceptual_Motor_Z-Score',
 'Motor-Perceptual_Motor_T-Score',
 'Motor-Perceptual_Motor_NCE']]
motor_domain.head(2)

Child_ID Gender Date_of_Birth  Age Location_-_Sub_Level_1  \
0    44879      F    2022-09-27  1.0              Southwest   
1    47795      F    2020-12-23  2.0      Greater Nashville   

                  Program_Label  Motor_Sum_of_Scaled_Scores  \
0  BDI-3 Eligibility Evaluation                        16.0   
1  BDI-3 Eligibility Evaluation                        31.0   

   Motor_Developmental_Quotient Motor_Percentile_Rank  \
0                          88.0                    21   
1                         102.0                    55   

  Motor_95%_Confidence_Interval  ... Motor-Perceptual_Motor_RS  \
0                         82-96  ...                       NaN   
1                        96-108  ...                      12.0   

  Motor-Perceptual_Motor_SS Motor-Perceptual_Motor_PR  \
0                       NaN                       NaN   
1                      11.0                        63   

  Motor-Perceptual_Motor_AE Motor-Perceptual_Motor_RDI  \
0                       NaN                        NaN   
1                        24                      95/90   

  Motor-Perceptual_Motor_CSS  Motor-Perceptual_Motor_CSS_90%_  \
0                        NaN                              NaN   
1                      480.0                        460 - 500   

   Motor-Perceptual_Motor_Z-Score Motor-Perceptual_Motor_T-Score  \
0                             NaN                            NaN   
1                            0.33                           53.0   

   Motor-Perceptual_Motor_NCE  
0                         NaN  
1                          57  

[2 rows x 50 columns]

In [32]:
cognitive_domain = evals[['Child_ID',
 'Gender',
 'Date_of_Birth','Age',
 'Location_-_Sub_Level_1',
 'Program_Label','Cognitive_Sum_of_Scaled_Scores',
 'Cognitive_Developmental_Quotient',
 'Cognitive_Percentile_Rank',
 'Cognitive_95%_Confidence_Interval',
 'Cognitive_RDI',
 'Cognitive_Z-Score',
 'Cognitive_T-Score',
 'Cognitive_NCE','Cognitive-Attention_and_Memory_Examiner',
 'Cognitive-Attention_and_Memory_Date_of_Testing',
 'Cognitive-Attention_and_Memory_RS',
 'Cognitive-Attention_and_Memory_SS',
 'Cognitive-Attention_and_Memory_PR',
 'Cognitive-Attention_and_Memory_AE',
 'Cognitive-Attention_and_Memory_RDI',
 'Cognitive-Attention_and_Memory_CSS',
 'Cognitive-Attention_and_Memory_CSS_90%',
 'Cognitive-Attention_and_Memory_Z-Score',
 'Cognitive-Attention_and_Memory_T-Score',
 'Cognitive-Attention_and_Memory_NCE',
 'Cognitive-Reasoning_/_Academic_Skills_Examiner',
 'Cognitive-Reasoning_/_Academic_Skills_Date_of_Testing',
 'Cognitive-Reasoning_/_Academic_Skills_RS',
 'Cognitive-Reasoning_/_Academic_Skills_SS',
 'Cognitive-Reasoning_/_Academic_Skills_PR',
 'Cognitive-Reasoning_/_Academic_Skills_AE',
 'Cognitive-Reasoning_/_Academic_Skills_RDI',
 'Cognitive-Reasoning_/_Academic_Skills_CSS',
 'Cognitive-Reasoning_/_Academic_Skills_CSS_90%',
 'Cognitive-Reasoning_/_Academic_Skills_Z-Score',
 'Cognitive-Reasoning_/_Academic_Skills_T-Score',
 'Cognitive-Reasoning_/_Academic_Skills_NCE',
 'Cognitive-Perception_and_Concepts_Examiner',
 'Cognitive-Perception_and_Concepts_Date_of_Testing',
 'Cognitive-Perception_and_Concepts_RS',
 'Cognitive-Perception_and_Concepts_SS',
 'Cognitive-Perception_and_Concepts_PR',
 'Cognitive-Perception_and_Concepts_AE',
 'Cognitive-Perception_and_Concepts_RDI',
 'Cognitive-Perception_and_Concepts_CSS',
 'Cognitive-Perception_and_Concepts_CSS_90%',
 'Cognitive-Perception_and_Concepts_Z-Score',
 'Cognitive-Perception_and_Concepts_T-Score',
 'Cognitive-Perception_and_Concepts_NCE']]
cognitive_domain.head(2)

Child_ID Gender Date_of_Birth  Age Location_-_Sub_Level_1  \
0    44879      F    2022-09-27  1.0              Southwest   
1    47795      F    2020-12-23  2.0      Greater Nashville   

                  Program_Label  Cognitive_Sum_of_Scaled_Scores  \
0  BDI-3 Eligibility Evaluation                            20.0   
1  BDI-3 Eligibility Evaluation                            30.0   

   Cognitive_Developmental_Quotient Cognitive_Percentile_Rank  \
0                             100.0                        50   
1                             100.0                        50   

  Cognitive_95%_Confidence_Interval  ... Cognitive-Perception_and_Concepts_RS  \
0                            90-110  ...                                  9.0   
1                            95-105  ...                                 11.0   

  Cognitive-Perception_and_Concepts_SS Cognitive-Perception_and_Concepts_PR  \
0                                 13.0                                   84   
1                                  7.0                                   16   

  Cognitive-Perception_and_Concepts_AE Cognitive-Perception_and_Concepts_RDI  \
0                                   12                                 98/90   
1                                   18                                 29/90   

  Cognitive-Perception_and_Concepts_CSS  \
0                                 411.0   
1                                 437.0   

   Cognitive-Perception_and_Concepts_CSS_90%  \
0                                  392 - 431   
1                                  421 - 453   

   Cognitive-Perception_and_Concepts_Z-Score  \
0                                          1   
1                                     – 1.00   

  Cognitive-Perception_and_Concepts_T-Score  \
0                                      60.0   
1                                      40.0   

  Cognitive-Perception_and_Concepts_NCE  
0                                    71  
1                                    29  

[2 rows x 50 columns]

In [33]:
bdi3_domain = evals[['Child_ID',
 'Gender',
 'Date_of_Birth', 'Age',
 'Location_-_Sub_Level_1',
 'Program_Label','BDI-3_Total_Sum_of_Scaled_Scores',
 'BDI-3_Total_Developmental_Quotient',
 'BDI-3_Total_Percentile_Rank',
 'BDI-3_Total_90%_Confidence_Interval',
 'BDI-3_Total_95%_Confidence_Interval',
 'BDI-3_Total_Z-Score',
 'BDI-3_Total_T-Score',
 'BDI-3_Total_NCE']]
bdi3_domain.head(2)

Child_ID Gender Date_of_Birth  Age Location_-_Sub_Level_1  \
0    44879      F    2022-09-27  1.0              Southwest   
1    47795      F    2020-12-23  2.0      Greater Nashville   

                  Program_Label  BDI-3_Total_Sum_of_Scaled_Scores  \
0  BDI-3 Eligibility Evaluation                              89.0   
1  BDI-3 Eligibility Evaluation                             128.0   

   BDI-3_Total_Developmental_Quotient BDI-3_Total_Percentile_Rank  \
0                                99.0                          47   
1                                99.0                          47   

  BDI-3_Total_90%_Confidence_Interval BDI-3_Total_95%_Confidence_Interval  \
0                              96-102                              95-103   
1                              97-101                              96-102   

  BDI-3_Total_Z-Score  BDI-3_Total_T-Score BDI-3_Total_NCE  
0              – 0.07                 49.0              49  
1              – 0.07                 49.0              49

plot/chart each broken down as a total and into subsets to see the curve/outliers

1. In which domains (and sub-domains) are children performing highest and lowest?

2. What is the pattern of children's scores? Do individual children tend to perform similarly across all domains or are there differences from domain to domain? Are these patterns similar for all children?

3. Is there a pattern of significant differences in children's scores in the sub-domains within a domain? For example, a gap between a child's expressive (talking) and receptive (understanding) language scores may indicate that additional evaluation is needed for that child's eligibility. A gap between these sub-domains for all or most children may indicate the BDI-3 tool may lack sensitivity in the language domain.

4. Does the child's age seem to impact their scoring?

5. Eligibility is currently based on domains. A child would be considered eligible based on a 40% delay in one area or 25% delay in two areas, as determined by Development Quotient (DQ) scores. A DQ of 70 or less indicates a 40% delay, and a DQ between 71 and 78 indicates a 25% delay. If TEIS added an option for eligibility based on the total score on the BDI-3 total score (DQ score of 70 or less), are there any children who would have qualified that did not qualify based on the current method?

6. Is there any difference in scoring noted based on evaluation type?

7. Is there any notable pattern of scoring by region? Note: TEIS contracts with three agencies for evaluations (one per grand region) as follows:  
* East TN, First TN and Southeast
* Greater Nashville, Upper Cumberland, and South Central
* Northwest, Southwest, and Memphis Delta

8. Is there any notable pattern of scoring by evaluator?